deflate.h

```diff
@@ -344,13 +345,11 @@ static inline void put_uint32_msb(deflate_state *s, uint32_t dw) {
  */
 static inline void put_uint64(deflate_state *s, uint64_t lld) {
 #if defined(UNALIGNED64_OK)
-    *(uint64_t *)(&s->pending_buf[s->pending]) = lld;
+    memcpy(&s->pending_buf[s->pending], lld, 8);
     s->pending += 8;
 #elif defined(UNALIGNED_OK)
-    *(uint32_t *)(&s->pending_buf[s->pending]) = lld & 0xffffffff;
-    s->pending += 4;
-    *(uint32_t *)(&s->pending_buf[s->pending]) = (lld >> 32) & 0xffffffff;
-    s->pending += 4;
+    memcpy(&s->pending_buf[s->pending], lld, 8);
+    s->pending += 8;
```

Comment:
This change doesn't make sense... First two cases would need to be combined for little-endian systems.

Reply:
Has this been resolved? Seems like if the compiler doesn't support 64 bit unaligned it can split it up into 2 x 32-bit reads or perhaps just do memcpy.